engine/runtime/src/test/scala/org/enso/compiler/test/pass/analyse/DataflowAnalysisTest.scala

```diff
@@ -1115,32 +1114,34 @@ class DataflowAnalysisTest extends CompilerTest {
 
       val depInfo = ir.getMetadata(DataflowAnalysis).get
 
-      val block     = ir.asInstanceOf[IR.Expression.Block]
-      val xBind     = block.expressions.head.asInstanceOf[IR.Expression.Binding]
-      val xBindName = xBind.name.asInstanceOf[IR.Name.Literal]
-      val xBindExpr = xBind.expression.asInstanceOf[IR.Literal.Number]
-
-      // The IDs
-      val blockId     = mkStaticDep(block.getId)
-      val xBindId     = mkStaticDep(xBind.getId)
-      val xBindNameId = mkStaticDep(xBindName.getId)
-      val xBindExprId = mkStaticDep(xBindExpr.getId)
-
-      // The info
-      val dependents   = depInfo.dependents
-      val dependencies = depInfo.dependencies
-
-      // The test for dependents
-      dependents.getDirect(blockId) should not be defined
-      dependents.getDirect(xBindNameId) shouldEqual Some(Set(xBindId))
-      dependents.getDirect(xBindExprId) shouldEqual Some(Set(xBindId))
-
-      // The test for dependencies
-      dependencies.getDirect(xBindId) shouldEqual Some(
-        Set(xBindNameId, xBindExprId)
-      )
-      dependencies.getDirect(xBindNameId) shouldEqual None
-      dependencies.getDirect(xBindExprId) shouldEqual None
+      val block = ir.asInstanceOf[IR.Expression.Block]
+      if (!block.expressions.head.isInstanceOf[IR.Error.Resolution]) {
```

Comment:
So if it is a failed resolution, we should report a failure?

Reply:
We are testing erroneous code. The old parser use to construct some `IR` which the test verified to have some structure and proper metadata attributes. The new parser doesn't produce the same structure, but rather generates syntax error `IR`.

I decided to modify the test to accept both variants as both seem to report some error to the user. As such the test succeeds if there is `IR.Error` or if the structure is attributed with expected (error indicating) metadata.